src/roi/Roi.ts

```diff
@@ -575,12 +576,15 @@ function getBorders(roi: Roi): { ids: number[]; lengths: number[] } {
       }
     }
   }
-  let ids: number[] = Array.from(surroudingIDs);
-  let borderLengths = ids.map((id) => {
+  let id: number[] = Array.from(surroudingIDs);
+  let borderLengths = id.map((id) => {
     return surroundingBorders.get(id);
   });
-  return {
-    ids,
-    lengths: borderLengths,
-  };
+  let borders = id.map((id, index) => {
+    return {
+      connectedID: id,
+      length: borderLengths[index],
```

Comment:
you don't need `borderLengths` variable here.

you can `surroundingBorders.get(id)` here